### ISSUE TITLE
Clean New Install - Can't Join Server

### ISSUE THREAD

**Issue by TheElectricCake** [AllTheMods/ATM-3#75]
I've had weird flakey problems in the past with not being able to join the server sometimes, and usually if I reinstalled ATM3 from scratch it'd let me in, which suggests that somehow my files are getting corrupted (I've run various tests on my system...It's not my HDD and I don't have any malware that'd explain it) or that something is being written/flagged as part of a mod that's causing this.

This time it happened on a brand new, clean install. WTH?!

[latest.txt](https://github.com/AllTheMods/ATM-3/files/1178430/latest.txt)

**Comment by jeffwcollins** [comment]
Dude, I gatta say, it's most likely either your router (read nat settings), pc (read network acceleration settings), or the server (but most likely not).  The log says you were disconnected by the remote host because it thinks you are already connected, meaning the server kicked you probably due to repeated connection attempts all at the same time.

**Comment by Bennyboy1695** [comment]
`NetworkDispatcher exception
java.io.IOException: An existing connection was forcibly closed by the remote host` - This error was at the bottom of your log, this usually occurs when ports aren't being forwarded correctly or firewall etc. This Reddit post should help you - https://www.reddit.com/r/Minecraft/comments/432m1f/fix_for_internal_exception_javaioioexception_an/

**Comment by TheElectricCake** [comment]
Unfortunately, this doesn't seem to apply to me. Turning off the NAT routing hasn't fixed it, and it ONLY happens in ATM 3, not in any other mudpack or on any other server.

**Comment by jeffwcollins** [comment]
I also see this:

17:45:21] [main/INFO]: Connecting to atm.g.akliz.net., 39034
[17:45:41] [Netty Client IO #1/INFO]: Server protocol version 2
[17:45:41] [Netty Client IO #1/INFO]: Attempting connection with missing mods [blockdrops, blur, controlling, ctgui, ctm, custommainmenu, defaultoptions, itemscroller, jeresources, modnametooltip, moreoverlays, reauth, resourceloader, toastcontrol] at SERVER

**Comment by Bennyboy1695** [comment]
@jeffwcollins Those are client side mods, that's why they say the server doesn't have them.

**Comment by TheElectricCake** [comment]
Here's the most recent log of the occurrence:


[latest.txt](https://github.com/AllTheMods/ATM-3/files/1294026/latest.txt)

**Comment by Bennyboy1695** [comment]
Does it only happen when your at the "spawn" ? As i saw you teleported there and then got the errors

**Comment by OrdinatorStouff** [comment]
I'm trying a fix for this with Java JVM args. It's possible that the JVM arg: `-Dfml.readTimeout=90` can mitigate these kinds of problems, BUT it must be enabled on both the server AND the client. I'm having our host AKLiz add it to the official ATM3 server but again, it would need to be added in your individual client's settings as well and there's nothing I can do from within the modpack "settings" to force that argument to be used.

I can at least add it as a default option on the ServerStart scripts for our server releases but not everyone uses our scripts to launch either.

Overall this is very borderline to "out of scope" as far as what can be done at a modpack level. Hopefully these settings helps for the players on the official server if nothing else.

**Comment by TheElectricCake** [comment]
Here's the log from the immediate crash on MC startup after adding the JVM arg:

[2017-09-14-2.txt](https://github.com/AllTheMods/ATM-3/files/1304843/2017-09-14-2.txt)

**Comment by OrdinatorStouff** [comment]
Nothing for a clue at all in there...
```
[15:31:49] [Netty Client IO #4/INFO]: Attempting connection with missing mods [*snip*] at SERVER
[15:31:49] [Session-Validator/INFO]: Session validation successful
[15:32:00] [main/INFO]: Stopping!
```

The SERVER side hasn't had the JVM arg applied yet maybe that's why? It's still pending from the host.

**Comment by TheElectricCake** [comment]
[2017-09-19-5.txt](https://github.com/AllTheMods/ATM-3/files/1316522/2017-09-19-5.txt)

Note that this is no longer a 'can't connect,' it's a 'random crash after teleporting and then frequently when I connect the server isn't syncing ("Waiting on server info..." in WAILA when I mouse over blocks, then I get disconnected.).

**Comment by Boomflex** [comment]
Legacy error - no longer applies.
